src/openapi/streaming/streaming.js

```diff
@@ -523,6 +525,83 @@ Streaming.prototype.disposeSubscription = function(subscription) {
     }
 };
 
+function createHandleSubscriptionReadyForUnsubscribe(subscriptions, resolve) {
+    return function() {
+        if (subscriptions.every((subscription) => subscription.isReadyForUnsubscribeByTag())) {
+            resolve();
+        }
+    };
+}
+
+function unsubscribeSubscriptionByTag(serviceGroup, url, tag, shouldDisposeSubscription) {
+    const subscriptionsToRemove = this.subscriptions.filter((subscription) =>
+        subscription.serviceGroup === serviceGroup &&
+            subscription.url === url &&
+            subscription.subscriptionData.Tag === tag
+        );
+
+    let allSubscriptionsReadyResolve;
+    const allSubscriptionsReady = new Promise((resolve) => {
+        allSubscriptionsReadyResolve = resolve;
+    });
+    const handleSubscriptionReadyForUnsubscribe = createHandleSubscriptionReadyForUnsubscribe(subscriptionsToRemove, allSubscriptionsReadyResolve);
+
+    subscriptionsToRemove.forEach((subscription) => {

```

Comment:
same here, the current "style" of this library is for loops rather than .forEeach